dcos/dcos-ui
# DCOS-43896: Progress indicator doesn't go away without a page reload

Update the breadcrumbs for the services when
instances or status change.

Closes https://jira.mesosphere.com/browse/DCOS-43896

## Testing
1. Create a service with multiple instances.
Example JSON:
```
{
  "id": "/new-service-1",
  "instances": 10,
  "portDefinitions": [],
  "container": {
    "type": "MESOS",
    "volumes": []
  },
  "cpus": 0.1,
  "mem": 128,
  "requirePorts": false,
  "networks": [],
  "healthChecks": [],
  "fetch": [],
  "constraints": [],
  "cmd": "sleep 1000;"
}
```
2. Start it and quickly open the service details for that service.
2.5. (Optional) Open services overview in another tab.
3. As soon as the service is running, it should have an updated breadcrumb in the service detail page.

## Trade-offs
1. I tried adding an integration test, but I couldn't figure out how to mock the actual scale so that we can see the changed status. Even in the test where we test the actual scale action, we don't really scale. If someone knows how this could work, that would be great. This is what I have so far.
In `ServiceDetail-cy` in the `Navigate to service page` context.
```
      it("updates breadcrumbs on status change", function() {
        cy.route({
          method: "PUT",
          url: /marathon\/v2\/apps\/\/net/,
          response: []
        });
        cy.visitUrl({
          url: "/services/detail/%2Fnet"
        });

        cy.get(".button-narrow").click();
        cy.get(".dropdown-menu-items")
          .contains("Scale")
          .click();
        cy.get(".form-group input[name='instances']").type("00");
        cy.get(".button")
          .contains("Scale Service")
          .click();
        cy.get(".status-bar").should("exist");
      });
```
2. While working on this, I noticed that pods don't have a status in the breadcrumbs for some reason.

## Dependencies
None.

## Screenshots
### Before
![Peek 2019-06-13 14-40](https://user-images.githubusercontent.com/40791275/59429842-538a9c00-8de9-11e9-8339-b831f7e84198.gif)
![Peek 2019-06-13 14-41](https://user-images.githubusercontent.com/40791275/59429847-55545f80-8de9-11e9-9522-b7d832958337.gif)

### After
![Peek 2019-06-13 14-30](https://user-images.githubusercontent.com/40791275/59429702-fee72100-8de8-11e9-96c0-50199948b1ce.gif)
![Peek 2019-06-13 14-37](https://user-images.githubusercontent.com/40791275/59429711-01497b00-8de9-11e9-9839-e61abebc793a.gif)
